bee-network/bee-autopeering/src/peering/update.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{
+    filter::NeighborFilter,
+    manager::{self, OutboundNeighborhood},
+    neighbor::{salt_distance, Neighbor},
+};
+
+use crate::{
+    delay::ManualDelayFactory,
+    discovery::manager::get_verified_peers,
+    local::Local,
+    peer::lists::ActivePeersList,
+    request::RequestManager,
+    server::ServerTx,
+    task::Repeat,
+    time::{self, MINUTE, SECOND},
+    NeighborValidator,
+};
+
+use std::time::Duration;
+
+/// Outbound neighborhood update interval if there are remaining slots.
+#[allow(clippy::identity_op)]
+pub(crate) const OPEN_OUTBOUND_NBH_UPDATE_SECS: Duration = Duration::from_secs(1 * SECOND);
+/// Outbound neighborhood update interval if there are no remaining slots.
+#[allow(clippy::identity_op)]
+const FULL_OUTBOUND_NBH_UPDATE_SECS: Duration = Duration::from_secs(1 * MINUTE);
+
+pub(crate) static OUTBOUND_NBH_UPDATE_INTERVAL: ManualDelayFactory =
+    ManualDelayFactory::new(OPEN_OUTBOUND_NBH_UPDATE_SECS);
+
+#[derive(Clone)]
+pub(crate) struct UpdateContext<V: NeighborValidator> {
+    pub(crate) local: Local,
+    pub(crate) request_mngr: RequestManager,
+    pub(crate) active_peers: ActivePeersList,
+    pub(crate) nb_filter: NeighborFilter<V>,
+    pub(crate) outbound_nbh: OutboundNeighborhood,
+    pub(crate) server_tx: ServerTx,
+}
+
+pub(crate) fn do_update<V: NeighborValidator + 'static>() -> Repeat<UpdateContext<V>> {
```

Comment:
hmmm.... I guess you say this because it's just repeatedly calling `update_outbound` (which updates the outbound neighborhood), and not updating itself? :thinking:

Any Suggestions?

Reply:
I'm saying this because it really only create a Box, it doesn't call update at all

Reply:
I understand what you're saying now, but the most obvious counter-argument would be async/await functions.

They technically just return `Future`s, which don't do anything by default....still we name the functions as if they would perform those operations. If we'ld apply your logic your logic consistently we'ld have to call them: "create_send_bytes_future" for example.

Reply:
`do_update` returns a function pointer to the method that actually performs the update in certain intervals, so similarly to async/await the code is not run immediately, but when some "driver/runtime" decides to do so.

Reply:
The whole point of the async/await is to make you consider async functions as sync functions so that there is no mental overhead. We don't have to deal with boxing futures ourselves etc, ... Here you box yourself so I would argue this is a very different situation. But I can understand, your call

Reply:
Alright...

Notice, that the method is not called "update", but instead "do_update". There is a semantic difference between the two. The "do_" prefix signals that we just give an "instruction" or "command" to perform the update, like a teacher telling his student to close the window ("do_close_window"). Exactly that's what's happening here as well: We return a "do_update" instruction the Box, so the method could also be called "get_do_update_instruction", but that's even more confusing I think.